core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala

```diff
@@ -157,12 +180,22 @@ object BigQueryPlugin {
           .getMethod("copyOf", classOf[java.util.Map[_, _]])
           .invoke(imClass, _)
           .asInstanceOf[ImmutableMap]
+      val of: ImmutableMap =
+        imClass
+          .getMethod("of")
+          .invoke(imClass)
+          .asInstanceOf[ImmutableMap]
     }
 
-    val from: (ImmutableMap.ImmutableMap, ImmutableMap.ImmutableMap, Configuration, Integer, SQLConf, String, Optional[StructType]) => SparkBigQueryConfig =
-      methodFrom
-        .invoke(clazz, _, _, _, _, _, _, _)
-        .asInstanceOf[SparkBigQueryConfig]
+    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
+      case (a, b, c, d, e, f, g, h, i) =>
+        if (methodFrom7.isDefined)
+          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
+        else if (methodFrom8.isDefined)
+          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+        else
+          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+    }
```

Comment:
yes, good advise for me, I've changed. please check again.

Reply:
I only need to case whether one of the three methods 7|8|9 exists, and then invoke the corresponding method according to the number of parameters.
Instead of having to define the three methods separately.